platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -40,6 +40,12 @@ The previous step will help you identify potential problems by issuing deprecati
 
 === Potential breaking changes
 
+==== `JavaCompile` tasks may fail when using a JRE even if compilation is not necessary
+
+The `JavaCompile` tasks may fail when using a JRE instead of a JDK in more cases. The toolchain resolution code was changed to enforce the presence of a compiler when being asked for one. However, the `java-base` plugin used the `JavaCompile` compiler to determine the default source and target compatibility when `sourceCompatibility`/`targetCompatibility` or `release` were not set. Now that the compiler cannot be resolved, this default causes the build to fail when only given a JRE, even if no compilation needs to occur (for example, with no sources).
```

Comment:
```suggestion
The `JavaCompile` tasks may sometimes fail when using a JRE instead of a JDK. 
This is due to changes in the toolchain resolution code, which enforces the presence of a compiler when one is requested. 
Previously, the `java-base` plugin used the `JavaCompile` task to determine the default source and target compatibility when `sourceCompatibility,` `targetCompatibility,` or `release` were not set. 
With the new enforcement, the absence of a compiler causes the build to fail when only a JRE is provided, even if no compilation is needed (e.g., in projects with no sources).
```

Reply:
I made some minor edits, like keeping the "`sourceCompatibility`/`targetCompatibility` or `release`" -- this is intentionally not phrased as 3 separate items, but a pair and another separate item, as both compatibilities are used and should generally be set together.